topology/probes/istio/verifiers.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package istio
+
+import (
+	"github.com/skydive-project/skydive/graffiti/graph"
+	"github.com/skydive-project/skydive/probe"
+)
+
+type verifierHandler func(g *graph.Graph) *resourceVerifier
+
+type resourceVerifier struct {
+	graph.DefaultGraphListener
+	g               *graph.Graph
+	verifiers       []func(*graph.Graph)
```

Comment:
I suggest for using a scalar verifier:

```
verify            func(*graph.Graph)
```

Reply:
What if a single resource has more than one verification procedure (as actually happens in most cases)?

Reply:
DONE